### ISSUE TITLE
Prevent sleep when printing (Feature Request)

### ISSUE THREAD

**Issue by dmopalmer** [repetier/Repetier-Host-Mac#46]
If the Mac Energy Saver control panel is set to sleep after a certain time interval of inactivity, it does so even if it is printing, causing the print run to be halted (with extruder heat on, etc.) leading to a failed print.

The program should prevent sleep during active printing, (but allow sleep when not running).  Apple developer Q&A 1340 tells you how.
http://developer.apple.com/library/mac/#qa/qa1340/_index.html

**Comment by repetier** [comment]
Thanks for the link. Indeed it could be helpful for some users. Will try to include it for next update.

**Comment by porjo** [comment]
+1 

I've just hit this issue.

**Comment by repetier** [comment]
You could print with our Repetier-Server whcih also runs on mac. It disables sleep while printing.

**Comment by porjo** [comment]
Thanks for the info. I didn't know Repetier-Server existed...

From the point of view of a home user with unsophisticated requirements, it would be better if this behaviour were integrated in the host application. It seems like quite a simple fix. My father (who is not computer savvy) had no clue that the reason his print jobs were being interrupted was because the Mac was going to sleep!

BTW, thankyou for this great utility!

**Comment by repetier** [comment]
Yes it is simple and my mac version in development has it already. But we are so busy that I can not say when we will release the next pure mac version. We currently hope to be able to publush the windows host for mac as well, but that needs to be tested and depends on mono version now available. We stopped doing that cause we had to many incompatibilities but hope that 3 years later in is better.

**Comment by dmopalmer** [comment]
I've forked Repetier-Host-Mac to fix a few things, but I don't see any commits here since Nov 2013.  Is your 'mac version in development' branch committed anywhere so I can base from it and avoid re-inventing wheels?

**Comment by repetier** [comment]
The host is not open source any more, so this repository is not updates and used any more.
